### ISSUE TITLE
b-form-group: horizontal layout works only in some cases

### ISSUE THREAD

**Issue by mariusa** [cdmoro/bootstrap-vue-3#697]
**Describe the bug**
According to docs
https://cdmoro.github.io/bootstrap-vue-3/components/FormGroup.html#horizontal-layout

all these should work:
```html
 <b-form-group
                                label-cols="4"
                                label-cols-lg="2"
                                label="Test 1"
                            >
                                <b-form-input></b-form-input>
                            </b-form-group>


                            <b-form-group
                                label-cols="4"
                                label="Test 2"
                            >
                                <b-form-input></b-form-input>
                            </b-form-group>



                            <b-form-group
                                label-cols="auto"
                                label="Test 3"
                            >
                                <b-form-input></b-form-input>
                            </b-form-group>
```

but only Test 1 has a horizontal layout.

Version: 0.3.4

**Comment by mariusa** [comment]
Related, maybe I misunderstand usage:
`label-align-sm="end"` works, but `label-align="end"` doesn't work. 

I was hoping that if no viewport sizes are specified, all apply.

**Comment by VividLemon** [comment]
> Related, maybe I misunderstand usage:

You're not. This is a bug

**Comment by mariusa** [comment]
Hi, note that the tests above still doesn't work on 0.3.7:

![2022-09-29_16-31](https://user-images.githubusercontent.com/1283842/193045261-b64a9b68-335e-4b6f-abb8-374e98b52489.png)

Thanks

**Comment by VividLemon** [comment]
My mistake, I thought this was only for label-align classes.

**Comment by VividLemon** [comment]
I have finalized the changes and I believe everything should be working. I have also included test cases for these two issues. This truly would not have been as big of an issue if not for the render functions 🤢 Why not just use jsx at that point? 

The issue stemmed from a mishandling of how to created the align classes, and how it created the col props. Align xs class would be resulted to text-xs-{whatever}, which is not a valid class -> it is text-{whatever}. Similarly, prop 'xs' is not valid for BCol prop, whereas the other breakpoints are. 'xs' prop is actually prop 'cols'. 

You are free to open the issue again if I still have not managed to fix it. Npm release 0.3.8 contains the fix if you would like to view it in your app.
